### ISSUE TITLE
Add edit button in contact diary where user selects entries

### ISSUE THREAD

**Issue by stefan-schoene** [corona-warn-app/cwa-app-ios#2196]
## Avoid duplicates

* [x] This issue has not already been raised before
* [x] If you are proposing a new feature, please do so in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)

## Current Implementation
Currently the only way to edit the list of locations and people is in the contact diary's main view via the context menu in the upper right corner.

## Suggested Enhancement

 
Basically all list views in iOS feature an edit button in the top corner so the user might expect one in this view as well, but then needs to go back to the parent view to change or delete entries. Therefore I would suggest adding this button for calling the edit view directly.
![IMG_0155](https://user-images.githubusercontent.com/78681069/110814846-2141f300-828a-11eb-804b-e06e3414fb90.jpeg)

## Expected Benefits

The user can directly edit all the entries in the same view where he also chooses them, making editing more quickly and bringing it on par with basically all list views in iOS.

**Comment by Ein-Tim** [comment]
Related Issue: https://github.com/corona-warn-app/cwa-wishlist/issues/334

**Comment by dsarkar** [comment]
@stefan-schoene Thanks for you suggestion.

---
Internal Tracking ID: [EXPOSUREAPP-3034](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-3034)

**Comment by cornelius-behrend** [comment]
We're on it together with @m-seidel

**Comment by dsarkar** [comment]
@cornelius-behrend @m-seidel You would like to provide a PR? If so, I will cross-check with developers. 
Thanks. Best wishes, DS

---
Corona-Warn-App Open Source Team

**Comment by Ein-Tim** [comment]
@cornelius-behrend 

AFAIK you can just start working (I see you already [did](https://github.com/m-seidel/cwa-app-ios/commit/ddb597b1a3fc3c97d3ddcb6ba9c5c94f645baf33)). 

The contributing document is outdated reg. this, contributors aren't assigned to issues anymore (or @dsarkar?)

**Comment by cornelius-behrend** [comment]
@dsarkar yes, we would like to add a PR for that if the issue was not already tackled. We would appreciate if you could check it :)

**Comment by dsarkar** [comment]
@cornelius-behrend @m-seidel OK, I will check if they would be interested in such a PR. However, they might not merge it, at the end, and this can have different reasons, e.g. quality of the code, rejected by the stakeholders, or other reasons. The are PRs from the community that have been merged, but not all of them if there was the need of major changes or quality issues or for other reasons. Stand by, I will com back to you. Thanks.

---
Corona-Warn-App Open Source Team

**Comment by cornelius-behrend** [comment]
@dsarkar thank you! We're doing this as part of the [Staffbase](https://www.staffbase.com) hackathon. So either way we learned something or we have contributed to the project which both is fine :)

**Comment by ndegendogo** [comment]
@cornelius-behrend @m-seidel
nice! Did you implement that for Android, or iOS, or even both?

**Comment by Ein-Tim** [comment]
I'm so free and post the link to the branch of @m-seidel & @cornelius-behrend here: https://github.com/m-seidel/cwa-app-ios/tree/feature/2196-add-edit-button-in-contact-diary

Also a huge thanks from my side! Hope that this makes it into the app!

**Comment by cornelius-behrend** [comment]
@Ein-Tim Thanks for posting the repo link! :) We're currently focussing on the iOS since we only have 2 days @ndegendogo

**Comment by dsarkar** [comment]
@ndegendogo @Ein-Tim @cornelius-behrend @m-seidel

Typically the stakeholders insist on feature parity for both OS. However, we will see, if there might be an exception, or if this PR triggers that the feature will be developed also in Android soon.

**Comment by Ein-Tim** [comment]
Some comments from my side:

1. A (new) user could think that the edit button is to remove the person/place for the day, and not from the whole contact diary. 
Thus, I would suggest that [the warning](https://user-images.githubusercontent.com/67682506/115063071-6097db00-9eeb-11eb-9c08-46c3ecd220bd.png) which now pops up when you delete a person or a place should definitely also be there. **This is something that should be implemented from the beginning on.**

2. We'll see more similar list views in the upcoming CWA version 2.0. The "My Check-Ins" screen (reachable by tapping on "Check In" on the Tab bar) & the "My QR Codes" screen (reachable by tapping on "Create QR Code" on the start screen).
These views should also have this edit button. **This is something that can also be implemented later.**

**Comment by dsarkar** [comment]
The story is currently described internally the following way:

As a user I would like to use commonly used ways of editing a person or place entry directly. 
- tap and hold on an entry to open a selection to edit or delete
- swipe to delete
- add a pencil icon to edit
- add a bin icon to delete

**Comment by cornelius-behrend** [comment]
@dsarkar

- [ ] tap and hold on an entry to open a selection to edit or delete
- [x] swipe to delete
- [ ] add a pencil icon to edit
- [x] add a bin icon to delete

I think 1 and 3 can be easily added as well. If you decide to have this as a PR, we will finish the task by adding this as well as fixing the remaining bugs and add tests.

**Comment by Ein-Tim** [comment]
@dsarkar @heinezen 

Could you give us feedback if the PR from @cornelius-behrend & @m-seidel would be accepted? It would be sad if this opportunity is not taken to get this create enhancement into the app.

**Comment by dsarkar** [comment]
@Ein-Tim see my latest comment in related issue: https://github.com/corona-warn-app/cwa-wishlist/issues/334. Nothing being worked on currently. As mentioned above, amongst others, stakeholders acceptance, quality of code and maintaining feature parity between Android and iOS are factors in such decisions.

**Comment by heinezen** [comment]
@Ein-Tim @cornelius-behrend @m-seidel

Just to be clear, the dev team would review a community PR for this and will likely merge it if it is accepted and passes the CI checks. Contributions are welcome here. The RKI and other stakeholders might require the same feature for iOS, but that shouldn't stop anyone from contributing for Android. 

----
Corona-Warn-App Open Source Team

**Comment by Ein-Tim** [comment]
@cornelius-behrend @m-seidel

Will you follow up with a PR? It would be a pity if these great changes just get lost!
